File: Java/Civitas/src/civitas/SorpresaPorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author arturo
 */
public class SorpresaPorCasaHotel extends Sorpresa{
    private Tablero tablero;
    private int cantidad;
    
    SorpresaPorCasaHotel(Tablero tablero, int cantidad, String texto){
        super(texto);
        this.tablero=tablero;
        this.cantidad=cantidad;
    }
    
    @Override
    public void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(this.jugadorCorrecto(actual, todos)){
            informe(actual, todos);
            todos.get(actual).modificarSaldo(this.cantidad*todos.get(actual).cantidadCasasHoteles()); 
        }
    }
    
    @Override
    public String toString(){
        return "Sorpresa: "+super.getTexto()+" Cantidad: "+this.cantidad+"\n";
    }
}

File: Ruby/Civitas/lib/sorpresa_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class SorpresaPorCasaHotel < Sorpresa
    def initialize(valor, texto)
      super()
      @valor=valor
      @texto=texto
      
      #puts "Sorpresa PorCasaHotel. \n"      
    end
    
    # Método sobrescrito
    def aplicar_a_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
      todos[actual].modificar_saldo(self.valor*todos[actual].propiedades.cantidad_casas_hoteles)
      end
    end

    def to_s
       "\n\t ¡SORPRESA! \n
       \n Has recibido: 
       \n #{@texto} \n"
    end
  end
end